pkgs/development/compilers/llvm/10/clang/default.nix

```diff
@@ -37,6 +38,8 @@ let
     ] ++ lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
       "-DLLVM_TABLEGEN_EXE=${buildLlvmTools.llvm}/bin/llvm-tblgen"
       "-DCLANG_TABLEGEN=${buildLlvmTools.libclang.dev}/bin/clang-tblgen"
+    ] ++ lib.optionals (stdenv.hostPlatform != stdenv.targetPlatform) [
+      "-DCLANG_DEFAULT_LINKER=${targetPackages.stdenv.cc.bintools}/bin/${stdenv.targetPlatform.config}-ld"
```

Comment:
Are you sure `targetPackages` is the right thing to use here?

`targetPackages` means `pkgsTargetTarget`, which is the packages which **run on** the `targetPlatform` and emit binaries for the `targetPlatform`.

When you're building a `(build==host)!=target` compiler, its linker will **run on** the `hostPlatform`

But I don't know clang very well so maybe I am misreading the meaning of `-DCLANG_DEFAULT_LINKER`.  @Ericson2314 would know for sure.

Holy smokes these treewide operations on `clang` are hard to review... the same diff repeated 12 times, and you just sorta cross your fingers and hope they aren't slightly different from each other.  Y'all really should consider pulling a https://github.com/NixOS/nixpkgs/pull/249707 on `clang`.

Reply:
Yep, I agree with @amjoseph-nixpkgs here.

I would expect rather `pkgsBuildTarget` to be used here, but maybe there's some splicing surprises… ?

Reply:
This might look suprising. Yes `targetPackages` is the packages that run on targetPlatform and emit binaries for the targetPlatform, but `targetPackages.stdenv` is something else. It is the environment that produces `targetPackages`. Now that `targetPackages` is the cross packages, `targetPackages.stdenv.cc.bintools` is the bintools that runs on host platform and emit target platform binaries.